java-frontend/src/main/java/org/sonar/java/model/JParserConfig.java

```diff
@@ -173,7 +173,7 @@ public void acceptAST(String sourceFilePath, CompilationUnit ast) {
             executionTimeReport.start(inputFile);
             Result result;
             try {
-              result = new Result(JParser.convert(javaVersion, inputFile.filename(), inputFile.contents(), ast));
+              result = new Result(JParser.convert(javaVersion.toString(), inputFile.filename(), inputFile.contents(), ast));
```

Comment:
Looking at `javaVersion.toString()` the reader could think there's a risk to return `"none"` to the lexer. In fact, it's not possible because of a previous call to `JParserConfig.effectiveJavaVersion`, but it's not obvious to understand. It would be easier to understand by moving the logic to JavaVersion and have a getter like `String javaVersion.effectiveVersion()`.